### PR TITLE
fix(observability-lib): improvements and fixes

### PR DESCRIPTION
**Pull request by Atrax1**
- Fix Panel ID needs to start at 1 instead of 0
- Add options for legend to add calculation functions
- Add mappings option for stat panel
- Fix gauge panel
- Add heatmap panel
- Add custom variable
- Refactor test that check json output
- Can create alert rules without having to attach it to panel
- Alerts can only be attached to timeseries panels as grafana integration on panels works only for timeseries
- ColorScheme for both threshold and lines added

### CODE REVIEW

File: observability-lib/Makefile [hunk 1]
```diff
@@ -12,4 +12,8 @@ lint:
 
 .PHONY: test
 test:
-	go test -v ./...
\ No newline at end of file
+	go test -v ./...
+
+.PHONY: regenerate
+regenerate:
+	go test -v ./dashboards/... -generate=1
```
Comment (jmank88) [inline observability-lib/Makefile:19]: Why `-v`?
```suggestion
	go test ./...

.PHONY: regenerate
regenerate:
	go test ./dashboards/... -generate=1
```

File: observability-lib/dashboards/k8s-resources/component_test.go [hunk 2]
```diff
@@ -10,6 +11,42 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+var generate = flag.Bool("generate", false, "generate test output")
```
Comment (jmank88) [inline observability-lib/dashboards/k8s-resources/component_test.go:14]: `update` is a common name for this:
```suggestion
var update = flag.Bool("update", false, "update golden test files")
```

File: observability-lib/dashboards/k8s-resources/component_test.go [hunk 3]
```diff
@@ -10,6 +11,42 @@ import (
...
+var fileOutput = "test-output.json"
```
Comment (jmank88) [inline observability-lib/dashboards/k8s-resources/component_test.go:15]: Constant?
```suggestion
const fileOutput "test-output.json"
```

File: observability-lib/Makefile (hunk 1 above)
Comment (Atrax1) [reply observability-lib/Makefile:19]: nice catch forgot to remove the flag after debugging some issues thanks

File: observability-lib/dashboards/k8s-resources/component_test.go (hunk 2 above)
Comment (Atrax1) [reply observability-lib/dashboards/k8s-resources/component_test.go:14]: Alright thanks for suggestion

File: observability-lib/dashboards/k8s-resources/component_test.go (hunk 3 above)
Comment (Atrax1) [reply observability-lib/dashboards/k8s-resources/component_test.go:15]: yes indeed

**Review by jmank88 — APPROVED** [review]

File: observability-lib/dashboards/core-node-components/component_test.go [hunk 4]
```diff
@@ -11,12 +12,51 @@ import (
...
+		errRemove := os.Remove(fileOutput)
+		if errRemove != nil {
+			t.Errorf("Error removing file: %v", errRemove)
+		}
+	}
+	file, errFile := os.Create(fileOutput)
+	if errFile != nil {
+		panic(errFile)
+	}
+	defer file.Close()
```
Comment (patrickhuie19) [inline observability-lib/dashboards/core-node-components/component_test.go:46]: could also consider using `t.Cleanup`. 

Have you considered making `TestGenerateFile` part of `TestNewDashboard`? IFIUC, both tests have the same input and output when calling `NewDashboard` and `GenerateJSON`

Comment (Atrax1) [reply observability-lib/dashboards/core-node-components/component_test.go:47]: NewDashboard inputs are different per dashboard 
the file generated has different content per dashboard

**Review by patrickhuie19 — APPROVED** [review]

File: observability-lib/dashboards/core-node-components/component_test.go (hunk 4 above)
Comment (patrickhuie19) [reply observability-lib/dashboards/core-node-components/component_test.go:47]: `TestGenerateFile` :

```

	testDashboard, err := corenodecomponents.NewDashboard(&corenodecomponents.Props{
		Name:              "Core Node Components Dashboard",
		MetricsDataSource: grafana.NewDataSource("Prometheus", ""),
		LogsDataSource:    grafana.NewDataSource("Loki", ""),
		Tested:            true,
	})
	if err != nil {
		t.Errorf("Error creating dashboard: %v", err)
	}
	json, errJSON := testDashboard.GenerateJSON()
	if errJSON != nil {
		t.Errorf("Error generating JSON: %v", errJSON)
	}
```

`TestNewDashboard`: 
```
func TestNewDashboard(t *testing.T) {
	t.Run("NewDashboard creates a dashboard", func(t *testing.T) {
		testDashboard, err := corenodecomponents.NewDashboard(&corenodecomponents.Props{
			Name:              "Core Node Components Dashboard",
			MetricsDataSource: grafana.NewDataSource("Prometheus", ""),
			LogsDataSource:    grafana.NewDataSource("Loki", ""),
			Tested:            true,
		})
		if err != nil {
			t.Errorf("Error creating dashboard: %v", err)
		}
		require.IsType(t, grafana.Dashboard{}, *testDashboard)
		require.Equal(t, "Core Node Components Dashboard", *testDashboard.Dashboard.Title)
		json, errJSON := testDashboard.GenerateJSON()
		if errJSON != nil {
			t.Errorf("Error generating JSON: %v", errJSON)
		}
```

look equivalent to me? what am i missing?

**Review by jmank88 — APPROVED** [review]

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**